### ISSUE TITLE
Lightbox filtering no longer working/isn't disallowing editing

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#59837]
[Pulling this out of a separate issue](https://github.com/WordPress/gutenberg/issues/59797#issuecomment-1994826807) and copying/pasting from @justintadlock  (let's make sure to create separate issues here for 6.5 problems). 

It looks like the filter itself works. However, the implementation in the editor might be broken. I'm testing against WP 6.5 RC2.

This combination isn't disallowing editing:

- `allowEditing`: `false`
- `enabled`: `true`

With those two settings, I'm still able to change the lightbox status of an existing Image block and a newly-added Image block in the editor.

Tested with filter:

```php
add_filter(
	'wp_theme_json_data_theme',
	fn($theme_json) => $theme_json->update_with([
		'version' => 2,
		'settings' => [
			'blocks' => [
				'core/image' => [
					'lightbox' => [
						'allowEditing' => false,
						'enabled'      => true
					]
				]
			]
		]
	])
);
```

I also tested this without the filter but setting it in `theme.json`:

```json
{
	"$schema": "https://schemas.wp.org/trunk/theme.json",
	"version": 2,
	"settings": {
		"blocks": {
			"core/image": {
				"lightbox": {
					"allowEditing": false,
					"enabled": true
				}
			}
		}
	}
}
```

CC: @artemiomorales 

This is a change in previous behavior from WordPress 6.4 and seems to be a regression and bug to me. I don't know if it's related to a change in the Lightbox toggle in the sidebar to the new location in the Link dropdown. https://github.com/WordPress/gutenberg/pull/57608 Or something else altogether.

**Comment by justintadlock** [comment]
Thanks for opening the ticket, @annezazu. I had some more testing to do before deciding this was a definite regression, but you beat me to it. :)

I've done some more testing with Twenty Twenty-Four. These are my `theme.json` settings:

```json
{
	"$schema": "https://schemas.wp.org/trunk/theme.json",
	"version": 2,
	"settings": {
		"blocks": {
			"core/image": {
				"lightbox": {
					"allowEditing": false,
					"enabled": true
				}
			}
		}
	}
}
```

As far as I can tell, this combination of settings is the only thing that differs between 6.4 an 6.5.

## With WordPress 6.4

With the above `theme.json` settings, the "Expand on click" toggle does not appear (user cannot edit lightbox state) in the sidebar. This would normally appear under the "Resolution" field in the sidebar.

![image](https://github.com/WordPress/gutenberg/assets/1816309/f01560a6-4685-49b6-baf2-8663fb8e8b1f)

## With WordPress 6.5

Using the same settings, the "Expand on click" button appears in the link dropdown in the toolbar (new location):

![image](https://github.com/WordPress/gutenberg/assets/1816309/c10a645c-bc08-4604-955f-4a6d29d69b2d)

**Comment by artemiomorales** [comment]
Hey all, I created a [PR to address the issue here](https://github.com/WordPress/gutenberg/pull/59890).

**Comment by gziolo** [comment]
The fix was released in [WordPress 6.5.2](https://wordpress.org/news/2024/04/wordpress-6-5-2-maintenance-and-security-release/).
